### ISSUE TITLE
Tricky bugs withs enum keyword (can be vacuum or lib openapi)

### ISSUE THREAD

**Issue by LasneF** [daveshanley/vacuum#339]
Hello , found a tricky bug 

here i am defining a meta schema simpleFieldSchema
so field must have type and description ,
and string can be restricted to an enum that is a array of values 

vacum raise a errors 
 Enum values must not have duplicate entry 
 and several 
 Enum values must respect the specified type

here it is

```
openapi: 3.0.0
info:
  title: dummy
  description: hello
  version: 1.0.0

servers:
  - url: /where


paths:
  /notsure:
    get:
      responses:
        '201':
          description: houston there is a pb

components:
  schemas:
    SimpleFieldSchema:
      description: Schema of a field as described in  JSON Schema draft 2019-09
      type: object
      required:
        - type
        - description
      properties:
        type:
          type: string
          enum:
            - string
            - number
        description:
          type: string
          description: A description of the property
        enum:
          type: array
          description: A array of describing the possible values
          items:
            type: string
          example:
            - yo
            - hello
```
notice that it went fine on others linters

**Comment by LasneF** [comment]
sorry typo, in the github between ` and 3` , looks ok now

**Comment by daveshanley** [comment]
I found the issue, very simple fix. `libopenapi` is indexing the property `enum` from the `SimpleFieldSchema`. This is not an enum, but it's a property named enum and the index is picking it up incorrectly. vacuum is then performing checks on a schema vs an actual enum, and it's getting upset.

The solution is to correctly have the index ignore properties named 'enum'. They are not enums :)

The fix will be available in the next release of vacuum.

**Comment by daveshanley** [comment]
Fixed in `v0.3.9`
